cmd/clusterctl/pkg/client/init.go

```diff
@@ -56,6 +53,14 @@ func (c *clusterctlClient) Init(options InitOptions) ([]Components, bool, error)
 		if len(options.BootstrapProviders) == 0 {
 			options.BootstrapProviders = append(options.BootstrapProviders, config.KubeadmBootstrapProviderName)
 		}
+		if len(options.ControlPlaneProviders) == 0 {
+			options.ControlPlaneProviders = append(options.ControlPlaneProviders, config.KubeadmControlPlaneProviderName)
+		}
+	}
+
+	// Before installing the providers, ensure the cert-manager WebHook is in place.
+	if err := cluster.CertManger().EnsureWebHook(); err != nil {
+		return nil, false, err
```

Comment:
It's not obvious to me why this moved, out of curiosity is there a particular reason this is better here?

Reply:
yes, I moved here so the users gets errors from provider components validation immediately (without waiting for cert manages to be installed)